functions/post-order.ts

```diff
@@ -202,7 +215,35 @@ async function getExchangeRate(usdAmount: number, fromCurrency: string, accessTo
   return responseJson as ExchangeRate;
 }
 
-function validateTransaction(txParsed: TransactionDescription, txReceipt: TransactionReceipt, chainId: number, giftCard: GiftCard): Response | void {
+function validateTransferTransaction(txParsed: TransactionDescription, txReceipt: TransactionReceipt, chainId: number, giftCard: GiftCard): Response | void {
+  const transferAmount = txParsed.args[1];
+
+  if (!ubiquityDollarAllowedChainIds.includes(chainId)) {
+    return Response.json({ message: "Unsupported chain" }, { status: 403 });
+  }
+
+  if (!isClaimableForAmount(giftCard, transferAmount)) {
+    return Response.json({ message: "Your reward amount is either too high or too low to buy this card." }, { status: 403 });
+  }
+
+  if (txParsed.functionFragment.name != "transfer") {
+    return Response.json({ message: "Given transaction is not a token transfer" }, { status: 403 });
+  }
+
+  if (txReceipt.to.toLowerCase() != ubiquityDollarErc20Address.toLowerCase()) {
+    return Response.json({ message: "Given transaction is not a Ubiquity Dollar transfer" }, { status: 403 });
+  }
+
+  if (txParsed.args[0].toLowerCase() != giftCardTreasuryAddress.toLowerCase()) {
+    return Response.json({ message: "Given transaction is not a token transfer to treasury address" }, { status: 403 });
+  }
+}
+
+function validatePermitTransaction(txParsed: TransactionDescription, txReceipt: TransactionReceipt, chainId: number, giftCard: GiftCard): Response | void {
+  if (!permitAllowedChainIds.includes(chainId)) {
+    return Response.json({ message: "Unsupported chain" }, { status: 403 });
```

Comment:
It seems like you're mocking network responses here locally?

Reply:
it's just returning response, not mocking

Reply:
My point is I'm confused why you're setting an HTTP status code for an object that originates within the client, and doesn't travel over the network.

Reply:
it is traveling over the network. The `functions` folder is considered backend and Cloudflare automatically deploys these endpoints when we deploy the website.